PlayCover/ViewModel/StoreVM.swift

```diff
@@ -27,16 +27,32 @@ class StoreVM: ObservableObject, @unchecked Sendable {
     }
     @Published var sourcesData: [SourceJSON] = [] {
         didSet {
-            sourcesApps.removeAll()
-            for source in sourcesData {
-                appendSourceData(source)
-            }
+            updateSourcesApps()
         }
     }
     @Published var sourcesApps: [SourceAppsData] = []
 
     private var resolveTask: Task<Void, Never>?
 
+    func enableSourceToggle(source: SourceData, value: Bool) {
+        if let index = sourcesList.firstIndex(of: source) {
+            sourcesList[index].isEnabled = value
+        }
+        updateSourcesApps()
+    }
+
+    func updateSourcesApps() {
+        sourcesApps.removeAll()
+        let enabledSources: [SourceJSON] = sourcesData.filter { sourceJSON in
```

Comment:
It would be nice to have the `enabledSources` filter be a function as this exact code is used multiple times.